test/solc/CommandLineParser.cpp

```diff
@@ -422,6 +420,52 @@ BOOST_AUTO_TEST_CASE(standard_json_mode_options)
 	BOOST_TEST(parsedOptions.value() == expectedOptions);
 }
 
+BOOST_AUTO_TEST_CASE(experimental_via_ir_invalid_input_modes)
+{
+	for (InputMode inputMode: {InputMode::StandardJson, InputMode::Assembler, InputMode::Linker})
+	{
+		vector<string> commandLine = {
+			"solc",
+			"--experimental-via-ir",
+			"file",
+		};
+
+		if (inputMode == InputMode::StandardJson)
+			commandLine.push_back("--standard-json");
+
+		if (inputMode == InputMode::Assembler)
+			commandLine.push_back("contract.yul");
+
+		if (inputMode == InputMode::StandardJson)
+		{
+			stringstream sout, serr;
+			optional<CommandLineOptions> parsedOptions = parseCommandLine(commandLine, sout, serr);
+
+			BOOST_TEST(sout.str() == "");
+			BOOST_TEST(serr.str() == "The option --experimental-via-ir is only supported in the compiler mode.\n");
+			BOOST_REQUIRE(!parsedOptions.has_value());
+		}
+		else
```

Comment:
This whole part is now unnecessary. The loop below already covers `--standard-json`.

Reply:
Removed